src/BotPlutusInterface/CardanoCLI.hs

```diff
@@ -236,27 +244,39 @@ signTx ::
   forall (w :: Type) (effs :: [Type -> Type]).
   Member (PABEffect w) effs =>
   PABConfig ->
+  Map PubKeyHash DummyPrivKey ->
   Tx ->
   [PubKey] ->
-  Eff effs ()
-signTx pabConf tx pubKeys =
-  callCommand @w $
-    ShellArgs
-      "cardano-cli"
-      ( mconcat
-          [ ["transaction", "sign"]
-          , ["--tx-body-file", txFilePath pabConf "raw" tx]
-          , signingKeyFiles
-          , ["--out-file", txFilePath pabConf "signed" tx]
-          ]
-      )
-      (const ())
+  Eff effs (Either Text ())
+signTx pabConf privKeys tx pubKeys =
+  let skeys = Map.filter (\case FromSKey _ -> True; FromVKey _ -> False) privKeys
+   in if all ((`Map.member` skeys) . Ledger.pubKeyHash) pubKeys
+        then callCommand @w $ ShellArgs "cardano-cli" opts (const (Right ()))
+        else do
+          let err =
+                Text.unlines
+                  [ "Not all required signatures have signing key files. Please sign and submit the tx manually:"
+                  , "Tx file: " <> txFilePath pabConf "raw" tx
+                  , "Signatories (pkh): "
+                      <> Text.unwords
+                        (map (encodeByteString . fromBuiltin . getPubKeyHash . Ledger.pubKeyHash) pubKeys)
+                  ]
+          printLog @w Warn (Text.unpack err)
+          pure $ Left err
```

Comment:
For manual signing, should the transaction error?
As far as a contract resulting in a manual sign goes, it did its job correctly

Reply:
Also, giving the tx here makes it pretty difficult to test contracts using this, i wonder if theres another way - perhaps a function specifically for under-signed transactions?

Reply:
This is a bit unfinished, I just wanted to push it asap, but certainly I want to find a better way to handle this. Demanding manual intervention is pretty annoying, also if a contract contains multiple txs, the whole workflow could break.
I am also thinking about a way to inject a callback function to handle the unsigned tx. I'll try to sketch up something in the following days...

Reply:
That could certainly work! Please do keep me in the loop on this :)